### PR TITLE
feat(config): support Zooz ZEN34, ZEN76, and ZEN77

### PR DESCRIPTION
**Pull request by mattstocum**
Adds support for new Zooz ZEN34, ZEN76, and ZEN77 switches.

### CODE REVIEW

**Comment by mattstocum** [comment]
These devices are not yet listed on products.z-wavealliance.org, so hopefully there won't be any conflicts. The association groups do not seem to be documented anywhere publicly, but I did confirm them with Zooz support.

**Comment by blhoward2** [comment]
No concerns, then. Did their support confirm everything? For whatever reason they don’t list their devices and whoever makes it for them sometimes does. I debated emailing them to see if they could send them en masse.

File: packages/config/config/devices/0x027a/zen76.json [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+				}
+			]
+		},
+		"3": {
+			"label": "Auto Turn-Off Timer",
+			"description": "Number of minutes to automatically turn off switch once it has been turned on (0 to disable).",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 65535,
+			"defaultValue": 0,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen76.json:145]: Add `"unit": "minutes"` please

File: packages/config/config/devices/0x027a/zen76.json [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"5": {
+			"label": "Auto Turn-On Timer",
+			"description": "Number of minutes to automatically turn on switch once it has been turned off (0 to disable).",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 65535,
+			"defaultValue": 0,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen76.json:157]: Add `"unit": "minutes"` please

File: packages/config/config/devices/0x027a/zen76.json [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+					"value": 1
+				}
+			]
+		},
+		"12": {
+			"label": "Smart Bulb Mode",
+			"description": "Enable or disable local on/off control. If enabled, you’ll only be able to control the connected light via Z-Wave. Scenes and other functionality will still be available through paddles.",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 3,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen76.json]:
```suggestion
			"maxValue": 2,
```
?

File: packages/config/config/devices/0x027a/zen76.json [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+					"value": 1
+				}
+			]
+		},
+		"16": {
+			"label": "Association Reports",
+			"description": "Report type for paddle control.",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 2,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen76.json]:
```suggestion
			"maxValue": 1,
```
?

File: packages/config/config/devices/0x027a/zen77.json [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+				}
+			]
+		},
+		"3": {
+			"label": "Auto Turn-Off Timer",
+			"description": "Number of minutes to automatically turn off switch once it has been turned on (0 to disable).",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 65535,
+			"defaultValue": 0,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen77.json:149]: Add `"unit": "minutes"` please

File: packages/config/config/devices/0x027a/zen77.json [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+			"writeOnly": false,
+			"allowManualEntry": true
+		},
+		"5": {
+			"label": "Auto Turn-On Timer",
+			"description": "Number of minutes to automatically turn on switch once it has been turned off (0 to disable).",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 65535,
+			"defaultValue": 0,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen77.json:161]: Add `"unit": "minutes"` please

File: packages/config/config/devices/0x027a/zen77.json [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+				}
+			]
+		},
+		"9": {
+			"label": "Ramp Rate Control",
+			"description": "Number of seconds for dimmer to reach full brightness or turn off when operated manually.",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 99,
+			"defaultValue": 1,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen77.json:198]: Add `"unit": "seconds"` please

File: packages/config/config/devices/0x027a/zen77.json [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+					"value": 1
+				}
+			]
+		},
+		"15": {
+			"label": "Smart Bulb Mode",
+			"description": "Enable or disable local on/off control. If enabled, you’ll only be able to control the connected light via Z-Wave. Scenes and other functionality will still be available through paddles.",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 3,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen77.json]: 2?

File: packages/config/config/devices/0x027a/zen77.json [hunk 9]
```diff
@@ -0,0 +1,407 @@
...
+				}
+			]
+		},
+		"16": {
+			"label": "Physical Dimming Speed",
+			"description": "The number of seconds it takes to get from 0% to 100% when pressing and holding the paddle.",
+			"valueSize": 1,
+			"minValue": 1,
+			"maxValue": 99,
+			"defaultValue": 5,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x027a/zen77.json:384]: Add `"unit": "seconds"` please

**Review by AlCalzone — CHANGES_REQUESTED** [review]

**Comment by mattstocum** [comment]
I think I got everything resolved. Sorry if the multiple commits are an issue, I know just enough git to get myself in to trouble and make a mess of commit history.

**Comment by mattstocum** [comment]
> No concerns, then. Did their support confirm everything? For whatever reason they don’t list their devices and whoever makes it for them sometimes does. I debated emailing them to see if they could send them en masse.

I got the config parameters for the 76 and 77 from https://www.support.getzooz.com/kb/article/549-zen76-s2-on-off-switch-700-advanced-settings/ and https://www.support.getzooz.com/kb/article/550-zen77-s2-dimmer-700-advanced-settings/. For the Zen34 I got everything from the manual.

I think the Zen26 and 27 have also been updated to have some of the new parameters that the 76 and 77 have with the 3.x firmware, but the online documentation hasn't been updated accordingly, but I haven't had time to test everything.

**Comment by AlCalzone** [comment]
> Sorry if the multiple commits are an issue

Not at all, I'm squash-merging every PR.

@blhoward2 let me know if your concerns are resolved, then I'll merge.

**Comment by blhoward2** [comment]
@AlCalzone I'm good

**Comment by anatwick** [comment]
I have been working with the guys over on discord with a Zooz Zen34 and there seems to be an issue with double tap's on the switch.  It reports it as central scene and a "double tap" triggers any automation set for the single tap.   Not sure exactly if i need to do something different for this or if it is some sort of bug.

**Comment by anatwick** [comment]
When listening for the command it appears to only recognize the first tap and nothing else  here is the automation yaml i was attempting for a double tap:   https://pastebin.ubuntu.com/p/Ddq3KzRYR8/

**Comment by blhoward2** [comment]
Please open a bug report so we can look into it. PRs are not the proper forum for troubleshooting.

**Pull request merged**